### PR TITLE
[TEL-304][TEL-306] Follow up pr for waiting area (ios)

### PR DESCRIPTION
**Pull request by ivanjiang5628**
## Description
- [Jira Ticket](https://janeapp.atlassian.net/browse/TEL-304)
- [Jira Ticket](https://janeapp.atlassian.net/browse/TEL-306)

Follow up PR for [TEL-281](https://github.com/janeapp/video-chat/pull/18)

- Update the pre-call message in `WaitingMessage` component for test my setup stage & waiting area.

- Add a close button to the `WaitingMessage` component to hide the pre-call info/message. 

- Add a hangup button to the waiting area component.

- If the JWT token has expired, the waiting area modal will display a "your appointment has expired" message and show "return to my account"/"Return to my Schedule" button to redirect the user back to jane.

- Refactor the code in the waiting area component.

### General PR Class
👁 = UX / UI improvement
🏗 = Refactor

- [ ] requires env configuration to be added in production
- [ ] js package changes<sup>1</sup>
- [x] more than 200 LOC changed in production files<sup>1</sup>
- [ ] includes a user-facing workflow change to an existing production feature (user muscle memory or pattern recognition will be affected)
- [ ] could prevent access to Jane Video (eg. cors, middleware, changes to auth system)
- [ ] affects a widely used component or piece of code
- [ ] I have a doubt - I want the RMT to review this. If possible, please elaborate your concerns in the risk assessment section.

<sup>1</sup> No need to explain these risk factors below

### Release Risk Assessment
Medium

### Demo Notes
Loom video:
https://www.loom.com/share/6960e2dc17394d069378babca655f580

## Code Review
Resource: [Dev Team Notion Page](https://www.notion.so/janeapp/Dev-Team-f06c6eb2ccca4066bc63fc1ac1bd2549)
Resource: [Code Review Checklist](https://www.notion.so/janeapp/Code-Review-checklist-2c510c527ac7470c902a5e8f25f9db3c)

- [x] I clearly explained the WHY behind the work, in the Description above

#### Design
- [x] I added instructions for how to test, in the QA section below
- [ ] I added specs for changes, or determined that none were required
- [ ] I demoed this to the appropriate person
- [x] I considered both mobile & desktop views, or that wasn't relevant

#### Code
- [x] I committed code with informative git messages
- [x] I wrote readable code, or added comments if it was complex
- [x] I performed a self-review of my own code
- [ ] I rebased my branch on the latest `master`

### Steps to Reproduce
Loom video:
https://www.loom.com/share/6960e2dc17394d069378babca655f580

1. Create an online appointment  -> go to the video chat mobile app -> video chat waiting area page.

2. Create an online appointment  -> click "test my setup" button -> video chat test audio/video page.

3. open an expired appointment session -> on the waiting area page the modal will show `Your appointment has expired` message and the `return to my account` button

## Screenshots
waiting area ui (patient side):

waiting area ui (practitioner side): 

Test my setup stage ui : 

token expired ui:

### CODE REVIEW

File: react/features/base/react/components/native/WaitingMessage.js [hunk 1]
```diff
@@ -37,11 +45,20 @@ class WaitingMessage extends Component<Props, State> {
...
             appointmentStartAt: '',
-            fadeAnim: new Animated.Value(0)
+            fadeAnim: new Animated.Value(0),
+            hideWaitingMessage: props.hideWaitingMessage
         };
         this.animatedValue = new Animated.Value(0);
     }
 
+    componentDidUpdate(props) {
+        if (props.hideWaitingMessage !== this.props.hideWaitingMessage) {
```
Comment (hollandmatt) [inline react/features/base/react/components/native/WaitingMessage.js:55]: Just wondering what we're trying to achieve here? This is the sort of thing that makes me worry about some tricky bugs later on 😉

File: react/features/base/react/components/native/WaitingMessage.js [hunk 2]
```diff
@@ -161,17 +172,35 @@ class WaitingMessage extends Component<Props, State> {
...
             inputRange: [ 0, .5, 1 ],
             outputRange: [ .1, 1, .1 ]
         });
 
         const image = <Image style={styles.watermark}
                              source={watermarkImg}/>;
-
-        return (<View style={[ styles.waitingMessageContainer ]}>
+        const backgroundColor = hideWaitingMessage ? 'transparent' : 'rgba(255,255,255,.3)';
+        const iphoneHasNotch = Platform.OS === 'ios' && this._getWindowHeight() > 811;
```
Comment (hollandmatt) [inline react/features/base/react/components/native/WaitingMessage.js:195]: Pity there isn't a better API for this... how does it work with iPads? (Or does our app not support iPad?)

**Review by hollandmatt — APPROVED** [review]
Looks good based on my React Native knowledge! Just had a couple of questions...

File: react/features/base/react/components/native/WaitingMessage.js (hunk 2 above)
Comment (ivanjiang5628) [reply react/features/base/react/components/native/WaitingMessage.js:195]: Good catch!👍 I will update the condition for the iPads.

File: react/features/base/react/components/native/WaitingMessage.js (hunk 1 above)
Comment (ivanjiang5628) [reply react/features/base/react/components/native/WaitingMessage.js:55]: You are right! I should remove/clean up this componentDidUpdate lifecycle.

**Comment by jngo-janeapp** [comment]
## Before Testing

- [ ] CI must be passing
- [ ] No conflicts
- [ ] No unresolved reviews/changes requested
- [ ] No additional changes/pushes from dev since the last round of QA
- [ ] Release note is sufficient for [#features](https://janeapp.slack.com/archives/C0KC1395G)
- [ ] Read through Notion / Jira story / Linear Issue / Slack Convo / Demo video

## Developer's Recommended Test Plan
- [ ] Create an online appointment -> go to the video chat -> video chat waiting area page.
- [ ] Create an online appointment -> click "test my setup" button -> video chat test audio/video page.
- [ ] Open the video chat with an expired session url

## Additional Thoughts on Testing
- [ ] TBD

## Mobile, Browsers, Jane Desktop
- [ ] Google Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Microsoft Edge
- [ ] Jane Desktop
- [ ] iPad Mini
- [ ] iPhone

## Regression Testing / Exploratory
(Consider related areas or areas where data persists Ie. Chart Templates/Intake Forms, Chart Entries/Export to PDF, Reports / Export to CSV/xls, Payment / Receipts / Superbill(US))
- Can you export a PDF from this area/function?
- Can this functionality be reused somewhere else in Jane? Or does it share components with other areas in Jane. (Canadian & US Insurance)
- Does this area have an inverse setting? Enabled/Disabled, Beta Feature Flag
- Does this area have multiple setting types? Ie. Intake Forms, Permissions/Privacy, User types
- Does this correspond to an area in Online Booking? (Booking an appointment, Entering a credit card and paying for an appointment)
- Is this related to Payment Methods? Jane Payments: Stripe, Payfirma, Square Terminal, Cash, Gift Cards, Credit Memos, Account Credit, Packages
- Could this be affected by Family Relationships?
- Could data be affected by Merging Patient Profiles?
- Could data be affected by rebooking/canceling appointments or changing treatments?
- Can this be Created, Updated, Archived/Unarchived or Deleted? (CRUD Test)
- Activity Report?
- Trigger a validation Error?

## Peer Review for Moderate Risk or Higher
- [ ] Peer Review

## Pass Story on QA Board / Mark Done in Jira/ Mark Scheduled for Release in Linear
- [ ] LGTM 🚀

**Pull request merged**